### ISSUE TITLE
synchrotrons and federation sender are getting behind on replication

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#7415]
I think we might now be processing replication data in smaller chunks than we were before, leading to more /sync wakeups and more outgoing federation requests. Mitigated by adding more synchrotrons but that doesn't solve the problem for federation.

**Comment by richvdh** [comment]
on closer inspection, this may be a problem but it's not what is actually causing cpu load on the synchrotrons and federation sender today.

**Comment by richvdh** [comment]
This might get mitigated by #7428 and #7429.

**Comment by erikjohnston** [comment]
Looking at the metrics for the federation sender over the last 30 days, some thoughts:

1. We don't see any increase CPU/DB usage of processing replication commands.
2. There are spikes of the in flight count of processing `RDATA` background jobs, but they seem to clear within a couple of minutes and happen relatively rarely so I'm not overly concerned.
3. We see a bunch of metrics increasing in tandem over the last few days:
 i. CPU usage
 ii. Federation worker stream getting behind
 iii. Increase number of EDUs being sent (mostly device lists and signing key updates, though also typing and receipts)
 iv. DB txn count and time for getting signing keys and device list updates for users.

Assuming that the increase in device list/signing key updates are legitimate (and not a manifestation of a replication bug), I think those increases dwarf any change caused by replication. An increase in traffic seems legitimate given Riot has just released apps supporting cross signing.

Broadly: I'm tempted to park this until we've fixed some of the perf woes caused by cross signing and then revisiting.

(I also took a flame graph and don't see anything obvious)
